var/spack/repos/builtin/packages/lua/package.py

```diff
@@ -58,24 +58,33 @@ class Lua(Package):
         destination="luarocks",
         placement='luarocks')
 
+    patch('https://git.archlinux.org/svntogit/packages.git/plain/trunk/liblua.so.patch?h=packages/lua&id=70abed4cfa74993348bf7afbbc70151a3d57cc01',
```

Comment:
maybe add this patch to Spack and also add another one from Homebrew: https://github.com/Homebrew/homebrew-core/blob/master/Formula/lua.rb (at the bottom)